keps/0041-dry-run.md

```diff
@@ -0,0 +1,66 @@
+---
+kep-number: 6
+title: Dry-run
+authors:
+  - "@apelisse"
+owning-sig: sig-api-machinery
+participating-sigs:
+  - sig-api-machinery
+  - sig-cli
+reviewers:
+  - "@lavalamp"
+  - "@deads2k"
+approvers:
+  - "@erictune"
+editor: apelisse
+creation-date: 2018-06-21
+last-updated: 2018-06-21
+status: implementable
+---
+# Kubernetes Dry-run
+
+Dry-run is a new feature that we intend to implement in the api-server. The goal
+is to be able to send requests to modifying endpoints, and see if the request
+would have succeeded (admission chain, validation, merge conflicts, ...) and/or
+what would have happened without having it actually happen. The response body
+for the request should be as close as possible to a non dry-run response.
+
+## Specifying dry-run
+
+Dry-run is triggered by setting the “dryRun” query parameter on modifying
+verbs: POST, PUT, PATCH and DELETE.
+
+This parameter is a string, working as an enum:
+- All: Everything will run as normal, except for the storage that won’t be
```

Comment:
@kubernetes/api-reviewers question here on what to call the dry run option to include all phases of the apiserver handling chain. Suggestions I've seen so far are `All`, `True`, and `*`